cosmwasm/enclaves/shared/contract-engine/src/contract_operations.rs

```diff
@@ -275,20 +275,46 @@ pub fn handle(
     };
 
     // There is no signature to verify when the input isn't signed.
-    // Receiving unsigned messages is only possible in Handle. (Init tx are always signed)
-    // The following messages go through handle but the data isn't signed:
-    //  * Replies from other sdk modules (WASM replies are signed)
-    if should_validate_sig_info {
+    // Receiving unsigned messages is only possible in Handle (Init tx are always signed).
+    // All of these scenarios go through here with HANDLE_TYPE_EXECUTE but the data isn't signed:
+    // - Plaintext replies (resulting from an IBC call) - doesn't have a msg.sender anyway
+    // - IBC WASM Hooks
+    // - (In the future:) ICA
+    //
+    // We do want to allow executing contracts with plaintext input via IBC
+    // while the sender of an IBC packet cannot be verified.
+    // And we don't want malicious actors using this enclvae setting to just
+    // impersonate any sender they want.
+    // => Therefore we'll use a null sender if it cannot be verified && the input is plaintext
+    let is_null_sender = if should_validate_sig_info {
```

Comment:
If we already know we're running with ibchooks (from the keeper execute param), why do we need another check here? Couldn't we just add another handle type and avoid all the logic here?

Reply:
Done in https://github.com/scrtlabs/SecretNetwork/commit/c11c681d1, LMK what you think